File: Mpress Unpacker/Unpacker.vb
Public Class Unpacker
    Public Shared Sub UnpackMpress(ByRef TheFile As String)
        Dim debugger As New NonIntrusive.NIDebugger
        Dim DumpOpts As New NonIntrusive.NIDumpOptions
        Dim ImportRec As New ImportReconstruction.ARImpRec
        Dim StartOpts As New NonIntrusive.NIStartupOptions
        Dim SearchOpts As New NonIntrusive.NISearchOptions

        With StartOpts
            .commandLine = ""
            .executable = TheFile
            .resumeOnCreate = False
        End With

       
        With debugger
            .Execute(StartOpts)
            .SingleStep()
            With SearchOpts
                .SearchString = "E9 ?? ?? ?? ?? ?? ?? ?? ?? 00 00 00 00 ?? ?? 00 00"
                .SearchImage = True
                .MaxOccurs = 1


            End With
            'E9 ?? ?? ?? ?? ?? ?? ?? ?? 00 00 00 00 ?? ?? 00 00
            Dim list() As UInteger = {}
            .SearchMemory(SearchOpts, list)
            If list.Length > 0 Then
                .SetBreakpoint(list(0))
            Else
                MsgBox("Fail are you sure its protected with Mpress?!")
                Exit Sub
            End If
TryAgain:
            .Continue()
            With SearchOpts
                .SearchString = "E9 ?? ?? FF FF ?? ?? ?? ?? FF 25 ?? ?? ?? 00 FF 25 ?? ?? ?? 00"
                .SearchImage = True
                .MaxOccurs = 1
            End With
            list = {}
            .SearchMemory(SearchOpts, list)
            If list.Length > 0 Then
                .SetBreakpoint(list(0))
                .Continue()
            Else
                .SetBreakpoint(.Context.Eip)
                .SingleStep()
                GoTo TryAgain
            End If
            .SingleStep()

            With DumpOpts
                .ChangeEP = True
                .EntryPoint = debugger.Context.Eip
                .OutputPath = Strings.Left(TheFile, TheFile.Length - 4) & "_dump.exe"
            End With
            .DumpProcess(DumpOpts)
            With ImportRec
                .Initilize(Application.StartupPath & "\")
                If .FixImports(debugger.Process.Id, DumpOpts.OutputPath, DumpOpts.EntryPoint, TheFile, True) = True Then
                    MsgBox("Successfully unpacked! Saved to: " & Environment.NewLine & .GetSavePath)
                Else
                    MsgBox("Import Reconstruction failed, Manually rebuild now!")
                End If
            End With
        End With
    End Sub
End Class

File: ASPackUnpacker/Unpacker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace ASPackUnpacker
{
    class Unpacker
    {
        /// <summary>
        /// The target that we are about to unpack...
        /// </summary>
        private string toBeUnpacked = string.Empty;
        /// <summary>
        /// Creates a handle to the class, and ready's the internal string holder...
        /// </summary>
        /// <param name="toUnpack">The target's filename...</param>
        public Unpacker(string toUnpack)
        {
            toBeUnpacked = toUnpack;
        }
        /// <summary>
        /// Unpacks the target with ease...
        /// </summary>
        /// <param name="myForm">The mainFrm</param>
        public void Unpack(mainFrm myForm)
        {
            NonIntrusive.NIStartupOptions opts = new NonIntrusive.NIStartupOptions();
            NonIntrusive.NIDebugger debugger = new NonIntrusive.NIDebugger();
            NonIntrusive.NIDumpOptions dumpOpts = new NonIntrusive.NIDumpOptions();
            NonIntrusive.NISearchOptions searchOpts = new NonIntrusive.NISearchOptions();

            uint[] list = new uint[1000];

            opts.executable = toBeUnpacked;
            opts.resumeOnCreate = false;

            dumpOpts.ChangeEP = true;
            dumpOpts.OutputPath = toBeUnpacked.Substring(0, toBeUnpacked.Length - 4) + "_dumped.exe";
            dumpOpts.PerformDumpFix = true;

            searchOpts.SearchString = "75 08 B8 01 00 00 00";
            searchOpts.SearchImage = true;
            searchOpts.MaxOccurs = 1;

            debugger.Execute(opts);

            debugger.SearchMemory(searchOpts, out list);
            if (list.Length > 0)
            {
                myForm.AddLog("Setting BP#1: " + (list[0] - debugger.ProcessImageBase).ToString("X8"));
                debugger.SetBreakpoint(list[0]).Continue().SingleStep(3);

                uint newOEP = debugger.Context.Eip - debugger.ProcessImageBase;
                dumpOpts.EntryPoint = newOEP;

                debugger.DumpProcess(dumpOpts);
                myForm.AddLog("OEP: " + newOEP.ToString("X8"));

                try
                {
                    Clipboard.Clear();
                    Clipboard.SetText(newOEP.ToString("X8"));
                }
                catch
                {
                    MessageBox.Show("Seems to have some problems clearing and setting the clipboard :(" + Environment.NewLine + "OEP: " + newOEP.ToString("X8"));
                }
                
                MessageBox.Show("Now fix the imports!");
                debugger.Detach().Terminate();
            }
            else
            {
                myForm.AddLog("Failed to find the OEP...");
                debugger.Detach().Terminate();
            }
        }
    }
}
